src/material/radio/radio.scss

```diff
@@ -153,7 +153,8 @@ $mat-radio-ripple-radius: 20px;
     opacity: 0.04;
   }
 
-  .mat-radio-button:not(.mat-radio-disabled).cdk-keyboard-focused & {
+  .mat-radio-button:not(.mat-radio-disabled).cdk-keyboard-focused &,
+  .mat-radio-button:not(.mat-radio-disabled).cdk-program-focused & {
```

Comment:
If I remember correctly, the reason why we aren't doing this at the moment is because the `FocusMonitor` incorrectly picks up focus that comes from clicking on the label of the radio button as `program`.

Reply:
Yeah that was one of the actual reasons why we had to remove the `cdk-program-focused`. 

@crisbeto you mentioned that you re-tested this and this seems like an IE only issue. That would be cool.. I still was under impression that we also ran into this w/ Chrome and Firefox.

It could be that https://github.com/angular/components/commit/a1aa9e7eb8a04603b36776d59cff637a10bf665f improved this as the label click no longer incorrectly causes the `program` focus origin.

Reply:
It looks like switching to check children for the focus monitor also helped. See: https://github.com/angular/components/commit/dda812264b5b839e8e2acb4607463caea827eb46